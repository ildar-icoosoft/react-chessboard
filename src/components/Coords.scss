.coords {
  font-size: 12px;
}

.ranks {
  flex-flow: column-reverse;
  height: 100%;
  right: -15px;
  top: 0;
  width: 12px;
}

.files {
  bottom: -15px;
  left: 0;
  text-align: center;
  flex-flow: row;
  width: 100%;
  height: 1.4em;
}
