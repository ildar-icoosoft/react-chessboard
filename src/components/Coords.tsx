import React, { FC } from "react";
import { PieceColor } from "../enums/PieceColor";
import { FILE_NAMES, RANK_NAMES } from "../constants/constants";
import css from "./Coords.scss";
import classNames from "classnames";

export interface CoordsProps {
  orientation?: PieceColor;
}

const getOrientationCssClass = (orientation: PieceColor) => {
  if (orientation === PieceColor.WHITE) {
    return "white";
  }
  return "black";
};

export const Coords: FC<CoordsProps> = ({ orientation = PieceColor.WHITE }) => {
  return (
    <>
      <div
        data-testid={"coords-ranks"}
        className={classNames(
          css.coords,
          css.ranks,
          css[getOrientationCssClass(orientation)]
        )}
      >
        {RANK_NAMES.map((item) => (
          <div key={item} className={css.coord}>
            {item}
          </div>
        ))}
      </div>
      <div
        data-testid={"coords-files"}
        className={classNames(
          css.coords,
          css.files,
          css[getOrientationCssClass(orientation)]
        )}
      >
        {FILE_NAMES.map((item) => (
          <div key={item} className={css.coord}>
            {item}
          </div>
        ))}
      </div>
    </>
  );
};
