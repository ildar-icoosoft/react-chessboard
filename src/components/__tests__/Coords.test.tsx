import React from "react";
import TestRenderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import { Coords } from "../Coords";
import { render } from "@testing-library/react";

jest.useFakeTimers();

describe("Square", () => {
  it("Snapshot", () => {
    const tree = TestRenderer.create(<Coords />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("DOM structure", () => {
    it("should contain data-testid coords-ranks", () => {
      const { queryByTestId } = render(<Coords />);
      expect(queryByTestId("coords-ranks")).toBeInTheDocument();
    });

    it("should contain data-testid coords-files", () => {
      const { queryByTestId } = render(<Coords />);
      expect(queryByTestId("coords-files")).toBeInTheDocument();
    });
  });
});
