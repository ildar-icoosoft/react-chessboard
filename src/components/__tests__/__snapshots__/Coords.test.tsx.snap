// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Snapshot 1`] = `
Array [
  <div
    className="coords ranks white"
    data-testid="coords-ranks"
  >
    <div
      className="coord"
    >
      1
    </div>
    <div
      className="coord"
    >
      2
    </div>
    <div
      className="coord"
    >
      3
    </div>
    <div
      className="coord"
    >
      4
    </div>
    <div
      className="coord"
    >
      5
    </div>
    <div
      className="coord"
    >
      6
    </div>
    <div
      className="coord"
    >
      7
    </div>
    <div
      className="coord"
    >
      8
    </div>
  </div>,
  <div
    className="coords files white"
    data-testid="coords-files"
  >
    <div
      className="coord"
    >
      a
    </div>
    <div
      className="coord"
    >
      b
    </div>
    <div
      className="coord"
    >
      c
    </div>
    <div
      className="coord"
    >
      d
    </div>
    <div
      className="coord"
    >
      e
    </div>
    <div
      className="coord"
    >
      f
    </div>
    <div
      className="coord"
    >
      g
    </div>
    <div
      className="coord"
    >
      h
    </div>
  </div>,
]
`;
